fedbiomed/node/fa_job.py

```diff
@@ -0,0 +1,135 @@
+        """Build error message for FA job failure."""
+        return ErrorMessage(
+            request_id=self._request_id,
+            researcher_id=self._researcher_id,
+            node_id=self._node_id,
+            node_name=self._node_name,
+            extra_msg=extra_msg,
+            errnum=errnum,
+        )
+
+    def _build_dataset(self) -> Dataset | ErrorMessage:
+        """Build dataset instance ready-to-use from dataset id."""
+        # dataset manager to get metadata about dataset, dlp and loading block
+        dataset_manager = DatasetManager(self._db_path)
+
+        # recover dataset entry
+        dataset_entry = dataset_manager.dataset_table.get_by_id(self._dataset_id)
+        if dataset_entry is None:
+            msg = f"Did not found proper data in local datasets on node={self._node_id}"
+            return self._build_error_msg(extra_msg=msg, errnum=ErrorNumbers.FB313.value)
+
+        # validate data type
+        data_type = dataset_entry.get("data_type")
+        if data_type not in REGISTRY_CONTROLLERS:
+            msg = f"Data type '{data_type}' not supported."
+            return self._build_error_msg(extra_msg=msg, errnum=ErrorNumbers.FB313.value)
+
+        # get controller parameters
+        controller_kwargs = {
+            "root": dataset_entry.get("path"),
+            **dataset_entry.get("dataset_parameters", {}),
+        }
+
+        # recover dlp if any
+        if "dlp_id" in dataset_entry:
+            dlp_metadata = dataset_manager.get_dlp_by_id(dataset_entry["dlp_id"])
+            try:
+                controller_kwargs["dlp"] = DataLoadingPlan().deserialize(*dlp_metadata)
+            except FedbiomedError as e:
+                msg = f"Cannot recover dlp on node={self._node_id}: {repr(e)}"
+                return self._build_error_msg(
+                    extra_msg=msg, errnum=ErrorNumbers.FB313.value
+                )
+
+        # build dataset instance
+        _, _, dataset_cls = REGISTRY_CONTROLLERS[data_type]
+        dataset = dataset_cls()
+        dataset.complete_initialization(controller_kwargs, DataReturnFormat.SKLEARN)
+
+        return dataset
+
+    def run(self) -> FAReply | ErrorMessage:
+        """Run FA job and return FAReply message or ErrorMessage in case of failure."""
+        # Retrieve dataset ready-to-use from self._dataset_id
+        dataset = self._build_dataset()
+        if isinstance(dataset, ErrorMessage):
+            return dataset
+
+        # TODO: implement FA job / needs to be adapted
+        _mean = dataset.mean()
+        output = {
+            "mean": [_mean[_feature] for _feature in self._fa_kwargs["mean"]],
+        }
+
+        return FAReply(
+            request_id=self._request_id,
+            researcher_id=self._researcher_id,
+            experiment_id=self._experiment_id,
+            fa_id=self._fa_id,
+            node_id=self._node_id,
+            node_name=self._node_name,
+            output=output,
+        )
```

Comment:
I think it could be nice to add `analytics_type` to the `FAReply`

```suggestion
        return FAReply(
            request_id=self._request_id,
            researcher_id=self._researcher_id,
            experiment_id=self._experiment_id,
            fa_id=self._fa_id,
            node_id=self._node_id,
            node_name=self._node_name,
            analytics_type="mean", # for now hardcoded but later can be set as defined constant variable value
            output=output,
        )
```

Reply:
It needs to be refined. Ideally, `output` should include all the required statistics. Hardcode specific names might constraint what is possible. I think `output` should be better defined (In my opinion statistics per feature could be a good idea: `{"feature1": {"mean": x, "std": y}, ...}`).
In any case it can be addressed in a more mature iteration.